### PR TITLE
Flush rewrite rules on plugin activation

### LINKED ISSUE #2174: Need to flush rewrite rules on plugin activation/deactivation
## Bug Description

Permalinks don't work out of the box when installing the plugin.

## Expected Behaviour

Stories permalinks should just work when installing and activating the plugin.

There should be no leftover rewrite rules when deactivating the plugin.

## Steps to Reproduce

1. Activate plugin on a fresh site
1. Publish a story
1. Notice that the rewrites don't work - you'll get a 404.

## Additional Context

- Plugin Version: `master`
- Operating System: -
- Browser: -

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

(Soft) Flushes rewrite rules on plugin activation and deactivation.

## Relevant Technical Choices

Ignore WP VIP specific PHPCS rule that seem to be built more for themes anyway

## To-do

N/A

## User-facing changes

N/A

## Testing Instructions

Test plugin activation & permalinks on a clean site. Use `wp rewrite` to check rewrite rules.

Rewrite rules **before** plugin **activation**:

```
$ wp rewrite list --match=stories/ffoo
+--------------------------+-------------------------------------------------+--------+
| match                    | query                                           | source |
+--------------------------+-------------------------------------------------+--------+
| (.?.+?)(?:/([0-9]+))?/?$ | index.php?pagename=$matches[1]&page=$matches[2] | page   |
| [^/]+/([^/]+)/?$         | index.php?attachment=$matches[1]                | post   |
+--------------------------+-------------------------------------------------+--------+
```

Rewrite  rules **after** plugin **activation**:

```
$ wp rewrite list --match=stories/ffoo
+----------------------------------+--------------------------------------------------+-----------+
| match                            | query                                            | source    |
+----------------------------------+--------------------------------------------------+-----------+
| stories/([^/]+)(?:/([0-9]+))?/?$ | index.php?web-story=$matches[1]&page=$matches[2] | web-story |
| (.?.+?)(?:/([0-9]+))?/?$         | index.php?pagename=$matches[1]&page=$matches[2]  | page      |
| [^/]+/([^/]+)/?$                 | index.php?attachment=$matches[1]                 | post      |
+----------------------------------+--------------------------------------------------+-----------+
```

Rewrite rules **after** plugin **deactivation**:

```
$ wp rewrite list --match=stories/ffoo
+--------------------------+-------------------------------------------------+--------+
| match                    | query                                           | source |
+--------------------------+-------------------------------------------------+--------+
| (.?.+?)(?:/([0-9]+))?/?$ | index.php?pagename=$matches[1]&page=$matches[2] | page   |
| [^/]+/([^/]+)/?$         | index.php?attachment=$matches[1]                | post   |
+--------------------------+-------------------------------------------------+--------+
```

----

Fixes #2174

### CODE REVIEW

File: includes/namespace.php [hunk 1]
```diff
@@ -46,6 +46,9 @@ function activate( $network_wide ) {
 		);
 	}
 
+	Story_Post_Type::init();
+	flush_rewrite_rules( false ); // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.flush_rewrite_rules_flush_rewrite_rules
```
Comment (spacedmonkey) [inline includes/namespace.php]: Should we wrap this in an if statement?
```
if ( !defined( '\WPCOM_IS_VIP_ENV' ) ){
    flush_rewrite_rules( false );
}
```

Comment (swissspidy) [reply includes/namespace.php]: Not familiar with WP.com VIP, so no idea what the best practice is here.

Comment (swissspidy) [reply includes/namespace.php]: Added now.

**Review by spacedmonkey — APPROVED** [review]
Looks good to me

**Pull request merged**